plugins/gradle/java/src/service/project/MavenRepositoriesProjectResolver.kt

```diff
@@ -15,6 +15,9 @@ import java.util.*
 
 
 class MavenRepositoriesProjectResolver: AbstractProjectResolverExtension() {
+
+  val processedRepositoryModels = IdentityHashMap<RepositoryModels, Unit>()
```

Comment:
Does the `IdentityHashMap` work as expected here?

 At first glance, different instances of `RepositoryModels` can contain duplicates - different  instances of `MavenRepositoryModel` that are logically equal, but have different references.